service/src/main/java/bio/terra/cbas/initialization/CloneRecoveryService.java

```diff
@@ -39,58 +41,77 @@ public void cloneRecovery() {
         cbasContextConfig.getWorkspaceId(),
         cbasContextConfig.getWorkspaceCreatedDate());
 
-    pruneCloneSourceWorkspaceHistory();
-  }
+    List<Method> clonedMethods =
+        methodDao.getMethods().stream().filter(this::isMethodCloned).toList();
 
-  @WriteTransaction
-  public void pruneCloneSourceWorkspaceHistory() {
-    List<Method> methods = methodDao.getMethods().stream().filter(this::isMethodCloned).toList();
+    List<MethodTemplateUpdateManifest> templateUpdateManifests =
+        clonedMethods.stream()
+            .map(m -> runSetDao.getRunSetsWithMethodId(m.methodId()))
+            .map(this::generateTemplateUpdateManifest)
+            .toList();
 
-    methods.forEach(this::updateMethodTemplate);
+    templateUpdateManifests.stream()
+        .map(MethodTemplateUpdateManifest::templateRunSets)
+        .forEach(rsList -> rsList.forEach(rs -> runSetDao.updateIsTemplate(rs.runSetId(), true)));
 
-    methods.forEach(this::cleanupClonedMethodHistory);
-  }
+    templateUpdateManifests.stream()
+        .map(MethodTemplateUpdateManifest::nonTemplateRunSets)
+        .forEach(rsList -> rsList.forEach(rs -> runSetDao.updateIsTemplate(rs.runSetId(), false)));
 
-  public Boolean isMethodCloned(Method m) {
-    return !m.originalWorkspaceId().equals(cbasContextConfig.getWorkspaceId());
+    clonedMethods.forEach(
+        method -> {
+          List<RunSet> methodRunSets = runSetDao.getRunSetsWithMethodId(method.methodId());
+          getRunSetsToDelete(methodRunSets).forEach(rs -> runSetDao.deleteRunSet(rs.runSetId()));
+          getRunsToDelete(methodRunSets).forEach(r -> runDao.deleteRun(r.runId()));
```

Comment:
Since you already know which run sets you're wanting to delete, why not just `delete from run where run_set_id = ...` right before deleting the run set itself? Could even be in the same Dao method (or at least transaction)?

Reply:
I was trying to reuse existing methods here, but a new Dao method is a good idea